hudi-cli/src/main/java/org/apache/hudi/cli/commands/SparkMain.java

```diff
@@ -159,6 +161,31 @@ public static void main(String[] args) throws Exception {
                   Boolean.parseBoolean(args[7]), Boolean.parseBoolean(args[8]));
           returnCode = 0;
           break;
+        case CLUSTERING_RUN:
+          assert (args.length >= 8);
+          propsFilePath = null;
+          if (!StringUtils.isNullOrEmpty(args[7])) {
+            propsFilePath = args[7];
+          }
+          configs = new ArrayList<>();
+          if (args.length > 8) {
+            configs.addAll(Arrays.asList(args).subList(8, args.length));
+          }
+          returnCode = cluster(jsc, args[1], args[2], args[3], Integer.parseInt(args[4]), args[5],
+              Integer.parseInt(args[6]), false, propsFilePath, configs);
+          break;
+        case CLUSTERING_SCHEDULE:
+          assert (args.length >= 6);
+          propsFilePath = null;
+          if (!StringUtils.isNullOrEmpty(args[5])) {
+            propsFilePath = args[5];
+          }
+          configs = new ArrayList<>();
+          if (args.length > 6) {
+            configs.addAll(Arrays.asList(args).subList(6, args.length));
+          }
+          returnCode = cluster(jsc, args[1], args[2], args[3], 1, args[4], 0, true, propsFilePath, configs);
```

Comment:
clusteringInstant use schedule generate will be more resonable. Because user set instant time may conflict with hudi. More information can see comments of https://github.com/apache/hudi/pull/2379

Reply:
Yes. The Clustering Instant here (args[4]) is generated at Line 57 of ClusteringCommand.java above. Users cannot set instant time for clustering.